src/GitVersion.MsBuild.Task.Tests/GitVersionTaskDirectoryTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.IO;
+
+using GitVersion;
+
+using LibGit2Sharp;
+
+using NUnit.Framework;
+
+[TestFixture]
+public class GitVersionTaskDirectoryTests : TestBase
+{
+    ExecuteCore executeCore;
+    string gitDirectory;
+    string workDirectory;
+
+
+    [SetUp]
+    public void CreateTemporaryRepository()
+    {
+        workDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        gitDirectory = Repository.Init(workDirectory)
+            .TrimEnd(Path.DirectorySeparatorChar);
+        executeCore = new ExecuteCore(new TestFileSystem());
+        Assert.NotNull(gitDirectory);
+    }
+
+
+    [TearDown]
+    public void Cleanup()
+    {
+        Directory.Delete(workDirectory, true);
+    }
+
+
+    [Test]
+    public void Finds_GitDirectory()
+    {
+        try
+        {
+            executeCore.ExecuteGitVersion(null, null, null, null, true, workDirectory, null);
+        }
+        catch (Exception ex)
+        {
+            // `RepositoryNotFoundException` means that it couldn't find the .git directory,
+            // any other exception means that the .git was found but there was some other issue that this test doesn't care about.
+            Assert.IsNotAssignableFrom<RepositoryNotFoundException>(ex);
```

Comment:
Can’t this, including the try/catch, be written as `Assert.That(() => …, Does.Not.Throw<RepositoryNotFoundException>());`?

Reply:
I took the GitVersionTask as a starting point for building GitVersion.MsBuild. The goal was to make the new task compatible with the old one. As for the Unit tests, they are the same as for GitVersionTask except they use the new Tasks.

Reply:
If we just took your `GitVersion.Msbuild` and just called it `GitVersionTask`, what would break?

Reply:
well eventually we can rename it back to `GitVersionTask` when it gets to a stable and tested level. As I said in the description I want to add some dockerfiles to validate it on all linux OSes Libgit2sharp is able to run, including .net core and mono, docs. I decided to create the PR first, give it a try and meanwhile work on the remaining stuff. As for the naming, to be honest I would keep `GitVersion.MsBuild` as it is clear it's about msbuild, `GitVersionTask` is not that suggestive, but I don't mind changing the name.

Reply:
I don’t mind the name change, I just think it’s enough with one solution. I completely agree with your approach, but at one point I would like to see one of `GitVersionTask` or `GitVersion.Msbuild` to go away so we only have one or the other.

Could we get this to preserve the Git history from `GitVersionTask` somehow? If we continue with this as is and remove everything related to the old `GitVersionTask`, we will lose all of its history. Ideas?

Reply:
Btw, GitVersionTask is still not able to resolve dependencies, so for now it will be broken. So either we go with GitVersion.MsBuild and remove GitVersionTask, or we just rename GitVersion.MsBuild to GitVersionTask

Reply:
I would think of a way of transforming GitversionTask into GitVersion.MsBuild keeping the history on the way.

Reply:
I'll try to do that and push again the changes.

Reply:
Perfect!